google/cloud/aiplatform/metadata/schema/base_artifact.py

```diff
@@ -0,0 +1,134 @@
+            Optional. The schema title used by the Artifact, defaults to "system.Artifact"
+        resource_name (str):
+            Optional. The resource name of the Artifact following the format as follows.
+            This is globally unique in a metadataStore:
+            projects/123/locations/us-central1/metadataStores/<metadata_store_id>/artifacts/<resource_id>.
+        uri (str):
+            Optional. The uniform resource identifier of the artifact file. May be empty if there is no actual
+            artifact file.
+        display_name (str):
+            Optional. The user-defined name of the Artifact.
+        schema_version (str):
+            Optional. schema_version specifies the version used by the Artifact.
+            If not set, defaults to use the latest version.
+        description (str):
+            Optional. Describes the purpose of the Artifact to be created.
+        metadata (Dict):
+            Optional. Contains the metadata information that will be stored in the Artifact.
+        state (google.cloud.gapic.types.Artifact.State):
+            Optional. The state of this Artifact. This is a
+            property of the Artifact, and does not imply or
+            capture any ongoing process. This property is
+            managed by clients (such as Vertex AI
+            Pipelines), and the system does not prescribe or
+            check the validity of state transitions.
+        **kwargs:
+            Optional. Additional Args that will be passed directly to the Artifact base method for backward compatibility.
+    """
+
+    ARTIFACT_PROPERTY_KEY_RESOURCE_NAME = "resourceName"
+    SCHEMA_TITLE = "system.Artifact"
+
+    def __init__(
+        self,
+        schema_title: Optional[str] = None,
+        resource_name: Optional[str] = None,
+        uri: Optional[str] = None,
+        display_name: Optional[str] = None,
+        schema_version: Optional[str] = None,
+        description: Optional[str] = None,
+        metadata: Optional[Dict] = None,
+        state: gca_artifact.Artifact.State = gca_artifact.Artifact.State.LIVE,
+        **kwargs,
+    ):
+
+        """Initializes the Artifact with the given name, URI and metadata."""
+        self.schema_title = BaseArtifactSchema.SCHEMA_TITLE
+        if schema_title:
+            self.schema_title = schema_title
+        self.resource_name = resource_name
+
+        self.resource_id = None
+        if resource_name:
+            # Temporary work around while Artifact.create takes resource_id instead of resource_name
+            # TODO: switch to using resouce_name only when create resouce supports it.
+            self.resource_id = resource_name.split("/")[-1]
+
+        self.uri = uri
+        self.display_name = display_name
+        self.schema_version = schema_version or constants._DEFAULT_SCHEMA_VERSION
+        self.description = description
+        self.metadata = metadata
+        self.state = state
+        self.kwargs = kwargs
```

Comment:
What's the purpose of this?

Reply:
I added kwargs per discussion on future proof / backward proofing the API.  we can remove this if it is not necessary or there is a different standard way that future/backward proofing is supported across the SDK.

Reply:
marking as resolved, kindly let me know if we should remove it for it.